include/glow/Base/Tensor.h

```diff
@@ -645,6 +658,69 @@ void dumpImpl(const Tensor *T, llvm::raw_ostream &os,
 void dumpImpl(const Tensor *T, unsigned maxNumElem);
 void dumpImpl(const Tensor *T);
 
+template <class ElemTy> class Handle;
+
+template <class ElemTy, bool IsConst>
+class HandleIterator : public std::iterator<std::forward_iterator_tag, ElemTy> {
```

Comment:
@putivsky mentioned this earlier I think, you could have pre-check whether the tensor has any alignments and if not this can just fall back to a basically just a pointer with no additional logic

Reply:
Done.